addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/PresenceDetection.java

```diff
@@ -306,24 +310,33 @@ public boolean performPresenceDetection(boolean waitForDetectionToFinish) {
         if (pingMethod != null) {
             detectionChecks += 1;
         }
-        if (arpPingMethod != ArpPingUtilEnum.UNKNOWN_TOOL) {
-            interfaceNames = networkUtils.getInterfaceNames();
-            detectionChecks += interfaceNames.size();
-        }
-
-        if (detectionChecks == 0) {
-            return false;
-        }
-
-        final ExecutorService executorService = getThreadsFor(detectionChecks);
-        this.executorService = executorService;
-
-        for (Integer tcpPort : tcpPorts) {
+        if (arpPingMethod != ArpPingUtilEnum.ELI_FULKERSON_ARP_PING_FOR_WINDOWS) {
```

Comment:
You are doing this at the wrong place. This is for the TCP services check. The next loop is for arp pings.

Reply:
Agreed, fixed!